shared-module/board/__init__.c

```diff
@@ -89,6 +89,8 @@ mp_obj_t common_hal_board_create_spi(void) {
     const mcu_pin_obj_t* mosi = MP_OBJ_TO_PTR(DEFAULT_SPI_BUS_MOSI);
     const mcu_pin_obj_t* miso = MP_OBJ_TO_PTR(DEFAULT_SPI_BUS_MISO);
     common_hal_busio_spi_construct(self, clock, mosi, miso);
+    // make sure lock is not held initially
+    common_hal_busio_spi_unlock(self);
```

Comment:
I wonder if this should go below before `spi_singleton = NULL` is set. That would handle a dangling try_lock when the display is in use.

Reply:
Testing on a PyBadge (which has an SPI display) showed that the proposed patch did not interfere with display functions. But, if you feel that a better place for it would be in reset_board_busses() I would be fine with that. Testing shows it works just as well there.

Reply:
Moved the unlock to reset_board_busses().